StarWars.Android/Resources/layout/activity_character_detail.xml

```diff
@@ -0,0 +1,276 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.core.widget.NestedScrollView 
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:local="http://schemas.android.com/apk/res-auto"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:fitsSystemWindows="true"
+        android:background="#000000">
+
+    <androidx.constraintlayout.widget.ConstraintLayout 
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical"
+            android:padding="32dp"
+            android:paddingTop="64dp"
+            android:clipToPadding="false">
+
+        <androidx.constraintlayout.widget.ConstraintLayout android:id="@+id/rowName" style="@style/StarJediLayout" 
```

Comment:
no no no
The main idea of ConstraintLayout is reduce nesting (that is equal to improve performance)
So you should use one single constraint layout as parent. And everything else constraint inside it

Reply:
You can also try Barriers and guidelines here